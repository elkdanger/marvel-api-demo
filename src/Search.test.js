import * as React from 'react'
import { shallow } from 'enzyme'
import SearchComponent from './Search'

it('can render the component', () => {
  const component = shallow(<SearchComponent />)

  expect(component.state().isDirty).toBe(false)
  expect(component.state().isValid).toBe(false)
  expect(component.state().searchTerm).toBe('')
})

it('changes input state when characters are entered', () => {
  const component = shallow(<SearchComponent />)

  component
    .find('input')
    .simulate('change', { currentTarget: { value: 'Character name' } })

  expect(component.state().searchTerm).toEqual('Character name')
  expect(component.state().isDirty).toBe(true)
  expect(component.state().isValid).toBe(true)
})

it('becomes invalid when the search term is entered and then removed', () => {
  const component = shallow(<SearchComponent />)

  component
    .find('input')
    .simulate('change', { currentTarget: { value: 'A name' } })
    .simulate('change', { currentTarget: { value: '' } })

  expect(component.state().isValid).toBe(false)
  expect(component.state().isDirty).toBe(true)
})
