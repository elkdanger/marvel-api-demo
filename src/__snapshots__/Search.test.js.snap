// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks the component against a snapshot 1`] = `
<div
  id="search"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Search for characters"
        type="text"
      />
      <span
        className="input-group-btn"
      >
        <button
          className="btn btn-primary"
          type="submit"
        >
          <span
            className="glyphicon glyphicon-search"
          />
        </button>
      </span>
    </div>
  </form>
</div>
`;
